ray-operator/apis/ray/v1/rayservice_types.go

```diff
@@ -57,6 +66,9 @@ type RayServiceSpec struct {
 	DeploymentUnhealthySecondThreshold *int32 `json:"deploymentUnhealthySecondThreshold,omitempty"`
 	// ServeService is the Kubernetes service for head node and worker nodes who have healthy http proxy to serve traffics.
 	ServeService *corev1.Service `json:"serveService,omitempty"`
+	// UpgradeStrategy represents the strategy used when upgrading the RayService. Currently supports `NewCluster` and `None`
+	// +kubebuilder:default:=NewCluster
```

Comment:
Defaulting `upgradeStrategy` to `NewCluster` seems to be a breaking change. For example, if users set `ENABLE_ZERO_DOWNTIME` to `false` and don't specify `upgradeStrategy`, zero-downtime upgrades were disabled in KubeRay v1.2.2 but are enabled with this PR.

Reply:
That is what I'm seeing as well. Should we not have a default value for `UpgradeStrategy` then? 
cc @andrewsykim What do you think?

Reply:
> Should we not have a default value for `UpgradeStrategy` then?

Having no default value sounds reasonable if there isn’t a better solution.

Reply:
Removed the defaulting.

Reply:
There needs to be defaulting to define the default upgrade strategy though? Why not have ENABLE_ZERO_DOWNTIME take precedence to preserve backwards compatibility?

Reply:
`ENABLE_ZERO_DOWNTIME` is an operator-level configuration, and `UpgradeStrategy` is a CR-level configuration. My current reasoning for why the operator-level config should take precedence is as follows:

1. KubeRay operator is managed by a ML infra team.
2. Custom resources are managed by application teams (e.g., recommendation systems, LLM, etc.).

The ML infra team will set a default configuration, i.e., `ENABLE_ZERO_DOWNTIME` in this case, for KubeRay that can work for most use cases.

However, in certain scenarios, some application teams may have rare use cases that require customizing their CRs. In these cases, they might want to set a different upgrade strategy for their RayService.

If the operator-level configuration takes precedence over the CR-level configuration, application teams would need to request changes to the KubeRay operator's configuration from the infra team. This could prevent these teams from sharing the KubeRay operator with other teams.

Does this make sense?

Reply:
> My current reasoning for why the operator-level config should take precedence is as follows:

Agreed operator-level config should take precedence for backwards compatibility, but UpgradeStrategy should still have a default when operator-level env var is not set.

Reply:
> UpgradeStrategy should still have a default when operator-level env var is not set right?

How about setting the default value in the KubeRay operator instead of OpenAPI, so that we can determine whether users have specified the field or not?

If users don't explicitly specify the value, we use the operator-level config.

Reply:
> How about setting the default value in the KubeRay operator instead of OpenAPI, so that we can determine whether users have specified the field or not?

Do you mean default the behavior only from the controller but not in the API? If we do this we probably want the field to be a pointer. 

But personally, I think defaulting the API is a better use experience. I'm assuming that most users have not set ENABLE_ZERODOWN_TIME=false. If they have, we can log a warning in KubeRay saying that the upgrade strategy API will be ignored until the envirionment variable is unset

Reply:
Discussed with @kevin85421 offline. 

I had mistakenly assumed that we would eventually remove support for ENABLE_ZERO_DOWNTIME with the addition of the new field. Since we plan to support both, we'll have to default `spec.upgradeStrategy` to `nil` and then change the default behavior based on whether ENABLE_ZERO_DOWNTIME is set or not.